### PR TITLE
DSBD-48: setting up some foundations for theming.

### PR DESCRIPTION
**Pull request by n3op2**
### What

Adding new interface for demoing against morello board. Made a few major changes

- theming, all in one place `/fixtures/themes.js`
- auto generator of tiles and context - `/fixtures/demos.js`
- a bit of clean up

### Screeen grabs
Screenshot 2022-08-30 at 16 07 26
Screenshot 2022-08-30 at 16 07 22
Screenshot 2022-08-30 at 16 07 17
Screenshot 2022-08-30 at 16 07 11
Screenshot 2022-08-30 at 16 07 13

### TODO
- [ ] - expand on cypress testing
- [x] - get an apple icon
- [x] - checkin with liam

### CODE REVIEW

File: src/components/Box.js [hunk 1]
```diff
@@ -1,62 +1,84 @@
...
-  outline: '2px solid #FFFFFF',
-  outlineOffset: '-10px',
-})
+  ...props,
+}))
 
-export default function Box(props) {
-  const { demo1, update } = React.useContext(Context)
-  const { isPasswordSet, renderModal } = demo1
+export default function Box(demo1) {
```
Comment (mattdean-digicatapult) [inline src/components/Box.js:17]: is `Box` now exclusive to `demo1`?

File: src/components/shared/Popup.js [hunk 2]
```diff
@@ -70,32 +68,31 @@ export default function PopupBox(props) {
     return null
   }
 
+  console.log(demo1)
```
Comment (mattdean-digicatapult) [inline src/components/shared/Popup.js:71]: remove `console.log`

**Review by mattdean-digicatapult — COMMENTED** [review]
Starter for 10 (I need to run this and see how it plays). One other thing is we should remove apple logo (trademark issues)

File: src/components/shared/Popup.js (hunk 2 above)
Comment (n3op2) [reply src/components/shared/Popup.js:71]: 🤦 Cypress was failing, wonder if we can upddate eslint to pick up on console.logs

**Comment by mattdean-digicatapult** [comment]
Close button seems to be broken?

**Comment by n3op2** [comment]
@mattdean-digicatapult possible.

**Comment by mattdean-digicatapult** [comment]
Also any ideas about:
Screenshot 2022-08-31 at 13 04 35

**Comment by mattdean-digicatapult** [comment]
> Also any ideas about: Screenshot 2022-08-31 at 13 04 35

I think this is due to the api in the docker compose being out of date, the UI refering to a different port to the compose and the wrong example being run

**Comment by jonmattgray** [comment]
Some of the new image assets (e.g. apple-logo.png) don't seem to be used anywhere, are they being added in anticipation of using in the future?

File: src/components/Box.js (hunk 1 above)
Comment (n3op2) [reply src/components/Box.js:17]: no

File: src/components/shared/Popup.js (hunk 2 above)
Comment (n3op2) [reply src/components/shared/Popup.js]: it has been removed

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
Some things I've noticed:

* integration with API does not work still. It's pointing at the wrong executable, not passing parameters, the docker compose is out of date etc. Lots of issues
* glitches on the "forward" button between examples (and it's on the left not the right which I find weird)
* state does not reset on close (you can only run the demo once)

All the above I can excuse as bugs which we can fix later to keep moving and get this merged. The apple logo has to go though (along with any other unused assets)

File: src/components/Header.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ export default function Header(props) {
                     renderModal: false,
                     isPasswordSet: false,
                     renderModalActions: true,
+                    outpsut: undefined,
```
Comment (mattdean-digicatapult) [inline src/components/Header.js:60]: typo

File: docker-compose.yml [hunk 4]
```diff
@@ -27,5 +27,5 @@ services:
     ports:
       - "3001:3000"
     environment:
-      - MORELLO_HOST=qemu
-      - MORELLO_PORT=19507
+      - DSBD_API_HOST=qemu
+      - DSBD_API_PORT=19507
```
Comment (mattdean-digicatapult) [inline docker-compose.yml:31]: these don't seem valid?

**Review by mattdean-digicatapult — COMMENTED** [review]
Couple of really small comments you may want to address

File: src/components/Header.js (hunk 3 above)
Comment (n3op2) [reply src/components/Header.js:60]: need a coffee

File: docker-compose.yml (hunk 4 above)
Comment (n3op2) [reply docker-compose.yml:31]: Shall I rename on the UI?

Comment (n3op2) [reply docker-compose.yml:31]: qemu comes from

```
services:
  morello-sdk-qemu:
    image: digicatapult/morello-sdk:v1.0.1
    container_name: qemu
    command: /bin/bash -c "
        qemu-system-morello
        -machine virt,gic-version=
```

**Review by mattdean-digicatapult — APPROVED** [review]
let's get this in and keep moving

**Pull request merged**